src/javascript/detectors/non_literal_regular_expression/non_literal_regular_expression.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// {fact rule=non-literal-regular-expression@v1.0 defects=1}
+var express = require('express')
+var app = express()
+function nonLiteralRegularExpressionNoncompliant() {
+    app.get("www.example.com", (req, res) => {
+        var re = new RegExp('ab+c')
+        // Noncompliant: user-controlled data passes into `test` for regex patterns.
+        var test = re.test(req.body.id)
+    })
+}
+// {/fact}
+
+
+// {fact rule=non-literal-regular-expression@v1.0 defects=0}
+var express = require('express')
+var app = express()
+function nonLiteralRegularExpressionCompliant() {
+    app.get("www.example.com", (req, res) => {
+        var re = new RegExp('ab+c')
+        // Compliant: hardcoded data passes into `test` for regex patterns.
+        var test = re.test('abccd')
```

Comment:
This `abccd` looks in-appropriate for the example since the need of `re.test()` is for `req.body.id` variable and here replacing by `abccd` does not explains why were we testing `req.body.id` variable in above.